src/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            _providers = new Dictionary<string, ILaunchSettingsProvider>
+            {
+                { ProjectLaunchSettingsProvider.CommandNameValue, new ProjectLaunchSettingsProvider() }
+            };
+        }
+
+        public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSettingsJsonContents, ref ICommand command, string profileName = null)
+        {
+            try
+            {
+                var model = JObject.Parse(launchSettingsJsonContents);
+                var profilesObject = model[ProfilesKey] as JObject;
+
+                if (profilesObject == null)
+                {
+                    return new LaunchSettingsApplyResult(false, LocalizableStrings.LaunchProfilesCollectionIsNotAJsonObject);
+                }
+
+                JObject profileObject;
+                if (profileName == null)
+                {
+                    profileObject = profilesObject
+                        .Properties()
+                        .FirstOrDefault(IsDefaultProfileType)?.Value as JObject;
+                }
+                else
+                {
+                    profileObject = profilesObject[profileName] as JObject;
+
+                    if (profileObject == null)
+                    {
+                        return new LaunchSettingsApplyResult(false, LocalizableStrings.LaunchProfileIsNotAJsonObject);
+                    }
+                }
+
+                if (profileObject == null)
+                {
+                    foreach (var prop in profilesObject.Properties())
+                    {
+                        var profile = prop.Value as JObject;
+
+                        if (profile != null)
+                        {
+                            var cmdName = profile[CommandNameKey]?.Value<string>();
+                            if (_providers.ContainsKey(cmdName))
+                            {
+                                profileObject = profile;
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                var commandName = profileObject?[CommandNameKey]?.Value<string>();
+
+                if (profileObject == null)
+                {
+                    return new LaunchSettingsApplyResult(false, LocalizableStrings.UsableLaunchProfileCannotBeLocated);
+                }
+
+                if (!TryLocateHandler(commandName, out ILaunchSettingsProvider provider))
+                {
+                    return new LaunchSettingsApplyResult(false, string.Format(LocalizableStrings.LaunchProfileHandlerCannotBeLocated, commandName));
+                }
+
+                return provider.TryApplySettings(model, profileObject, ref command);
+            }
+            catch (Exception ex)
+            {
+                return new LaunchSettingsApplyResult(false, string.Format(LocalizableStrings.UnexpectedExceptionProcessingLaunchSettings, ex.Message));
```

Comment:
This is only slightly better than before. It drives me up the wall to get only the exception message for cases that are *bugs*. "Unexpected exception processing launch settings: Object reference not set to an instance of an object." or "Unexpected exception processing launch settings: Sequence contains no elements." is still a pain to diagnose. Errors like this should be for things we expect to happen, not for bugs in our code. Let the bugs go.